src/coprocessor/codec/mysql/time/tz.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::str::FromStr;
+
+use chrono::*;
+use chrono_tz;
+
+/// A time zone represented by either offset (i.e. +8) or name (i.e. Asia/Shanghai). In addition,
+/// local time zone is also valid.
+#[derive(Clone, Copy)]
+pub enum Tz {
+    /// A time zone specified by offset seconds.
+    Offset(FixedOffset),
+
+    /// A time zone specified by name (in IANA database).
+    Name(chrono_tz::Tz),
+
+    /// A local time zone.
+    Local(Local),
+    // chrono also has a offset kind `chrono::offset::Utc`, but we only use UTC in tests,
+    // so we don't introduce an extra enum item for it.
+}
+
+impl Tz {
+    /// Constructs a time zone from the offset in seconds.
+    pub fn from_offset(secs: i64) -> Option<Self> {
+        FixedOffset::east_opt(secs as i32).map(Tz::Offset)
+    }
+
+    /// Constructs a time zone from the name. If the specified time zone name is `system`,
+    /// a local time zone will be constructed.
+    pub fn from_tz_name(name: &str) -> Option<Self> {
+        if name.to_lowercase() == "system" {
+            Some(Tz::local())
```

Comment:
Should we consider if tidb and tikv do not share the same timezone?

Reply:
Seems that at TiDB side it is difficult to push down the actual time zone name for now, because Golang's time library does not support this feature. We need to keep this functionality.

Reply:
I worry this will cause errors somehow.

Reply:
Agree! But we don't have better solutions for now.

Reply:
I will add a TODO here.

Reply:
Oh, it should be a TODO for TiDB :)